src/main/java/Tasks.java

```diff
@@ -0,0 +1,173 @@
+import java.io.*;
+import java.util.ArrayList;
+
+public class Tasks implements Serializable {
+    private static ArrayList<Task> tasks;
+    private static final String FILE_PATH = "src/data/taskList.txt";
+    public Tasks() {
+        // Load tasks from file on startup
+        loadTasks();
+    }
+
+    private static void loadTasks() {
+        try (ObjectInputStream ois = new ObjectInputStream(new FileInputStream(FILE_PATH))) {
+            tasks = (ArrayList<Task>) ois.readObject();
+        } catch (IOException | ClassNotFoundException e) {
+            // Handle file not found or corrupted data file
+            //System.err.println("Error loading tasks: " + e.getMessage());
```

Comment:
Would be good to remove deadcode. However I think in this case you might actually want it to be there to tell the user what happened!  
  
https://nus-cs2103-ay2324s2.github.io/website/se-book-adapted/chapters/codeQuality.html#delete-dead-code